fundamentals/android/auth-anonymous/Notes/app/src/main/java/com/notes/app/screens/sign_up/SignUpViewModel.kt

```diff
@@ -38,11 +38,19 @@ class SignUpViewModel @Inject constructor(
 
     fun onSignUpClick(openAndPopUp: (String, String) -> Unit) {
         launchCatching {
-            if (_password.value != _confirmPassword.value) {
+            if (!_email.value.isValidEmail()) {
+                throw Exception("Invalid email format")
```

Comment:
Should we try a more specific exception type? Maybe `IllegalArgumentException` ?

Reply:
It makes sense, good catch! Updated the exception type.